components/event-publisher-proxy/pkg/sender/jetstream/jetstream.go

```diff
@@ -58,48 +60,60 @@ func NewSender(ctx context.Context, connection *nats.Conn, envCfg *env.NATSConfi
 	return &Sender{ctx: ctx, connection: connection, envCfg: envCfg, opts: opts, logger: logger}
 }
 
-// ConnectionStatus returns nats.Status for the NATS connection used by the Sender.
+// ConnectionStatus returns nats.code for the NATS connection used by the Sender.
 func (s *Sender) ConnectionStatus() nats.Status {
 	return s.connection.Status()
 }
 
 // Send dispatches the event to the NATS backend in JetStream mode.
 // If the NATS connection is not open, it returns an error.
-func (s *Sender) Send(_ context.Context, event *event.Event) (sender.PublishResult, error) {
+func (s *Sender) Send(_ context.Context, event *event.Event) sender.PublishError {
 	if s.ConnectionStatus() != nats.CONNECTED {
-		return nil, ErrNotConnected
+		return ErrNotConnected
 	}
 
 	jsCtx, jsError := s.connection.JetStream()
 	if jsError != nil {
-		return nil, jsError
+		return common.ErrClientNoConnection
```

Comment:
should we log the jsError object here? or wrap it with the returned error?

Reply:
yeah, logging is probably a good idea

Reply:
Do we log here now?

Reply:
yes :)